library/src/main/java/com/novoda/downloadmanager/lib/DownloadsUriProvider.java

```diff
@@ -0,0 +1,88 @@
+package com.novoda.downloadmanager.lib;
+
+import android.net.Uri;
+
+class DownloadsUriProvider {
+
+    private final Uri publiclyAccessibleDownloadsUri;
+    private final Uri downloadsByBatchUri;
+    private final Uri allDownloadsUri;
+    private final Uri batchesUri;
+    private final Uri contentUri;
+
+    public static DownloadsUriProvider getInstance() {
+        return LazyInitialisationHelper.INSTANCE;
+    }
+
+    private static class LazyInitialisationHelper {
+        private static final DownloadsUriProvider INSTANCE = newInstance();
```

Comment:
What do you mean?

The static lazy initialisation is a standard procedure https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom